src/QueryM.purs

```diff
@@ -393,9 +393,22 @@ addressToOgmiosAddress :: Address -> JsonWsp.Address
 addressToOgmiosAddress = addressBech32
 
 -- If required, we can change to Either with more granular error handling.
--- | Gets utxos at an (internal) Address in terms of (internal) Transaction.Types
+-- | Gets utxos at an (internal) Address in terms of (internal) Transaction.Types.
+-- Results may vary depending on Wallet type.
 utxosAt :: Address -> QueryM (Maybe Transaction.UtxoM)
-utxosAt = addressToOgmiosAddress >>> getUtxos
+utxosAt addr = asks _.wallet >>= maybe (pure Nothing) (utxosAtByWallet addr)
+
+-- Add more wallet types here:
+utxosAtByWallet
+  :: Address -> Wallet -> QueryM (Maybe Transaction.UtxoM)
+utxosAtByWallet addr (Nami _) = utxosAtNami addr
+-- Unreachable but helps build when we add wallets, most of them shouldn't
+-- require any specific behaviour.
+utxosAtByWallet addr _ = utxosAt'' addr
+
+-- Gets utxos at an (internal) Address in terms of (internal) Transaction.Types.
+utxosAt'' :: Address -> QueryM (Maybe Transaction.UtxoM)
```

Comment:
I'm not a big fan of multiple primes, even for local bindings, but especially for top-level definitions I find it a little confusing. Could you think of a more descriptive name, or perhaps move some of the top-level `utxosAt...` effects to a `where` clause?